doc/sphinx/language/core/records.rst

```diff
@@ -55,31 +63,36 @@ synonym :cmd:`Structure`.
 
    In :n:`@record_field`:
 
+     :n:`@attribute`, if specified, can only be :n:`canonical`.
+
      :n:`@name` is the field name.  Since field names define projections, you can't
-     reuse the same field name in two different records in the same scope.
+     reuse the same field name in two different records in the same scope.  This
+     :ref:`example <reuse_field_name>` shows how to use the reuse the same field
+     name in multiple records.
 
-     :n:`field_type` can be omitted only when the type of the field can be inferred
+     :n:`@field_spec` can be omitted only when the type of the field can be inferred
      from other fields. For example: the type of :n:`n` can be inferred from
      :n:`npos` in :n:`Record positive := { n; npos : 0 < n }`.
 
      :n:`| @natural`
        Specifies the priority of the field.  It is only allowed in :cmd:`Class` commands.
 
-   In :n:`@field_type`:
+   In :n:`@field_spec`:
 
-     :n:`{+ @binder } : @of_type` is equivalent to :n:`forall {+ @binder } , @of_type`
-     and similarly for the other forms of :n:`@field_type`.
+     - :n:`{+ @binder } : @of_type` is equivalent to
+       :n:`forall {+ @binder } , @of_type`
 
-     .. todo PR: is the above correct for "binder := term"?
+     - :n:`{* @binder } := @term` is equivalent to
+       :n:`fun {* @binder } => @term`
```

Comment:
This should be ``:n:` := fun {* @binder } => @term` ``

Reply:
With `:=` or (like the other two) `:`?

Also to be precise, I'll use `{+ binder }` in each of these:

![image](https://user-images.githubusercontent.com/1253341/153304381-5071c82f-d479-463a-a992-751bf1af3821.png)

versus `{* binder }` in the grammar:

![image](https://user-images.githubusercontent.com/1253341/153304468-a049aa9e-14f5-4ed1-a421-bdf26b25ccad.png)

Reply:
Indeed, this is only when there is at least one binder.